### PR TITLE
Ingredients by location

### PR DESCRIPTION
**Pull request by Sgalvin36**
## Scope

Added an additional parameter for the show action on the recipe controller. If provided with a location parameter then the api will make a call to the Kroger api to get updated prices based off the store location provided and return those prices with the rest of the recipe data from our database

### CODE REVIEW

**Comment by Sgalvin36** [comment]
Also added the functionality of an update to already existing recipes that we can access externally with a call to the recipe builder.

File: app/controllers/api/v1/recipe_builder_controller.rb [hunk 1]
```diff
@@ -12,6 +12,17 @@ def create
...
+        recipe = Recipe.find(params[:id])
+        builder = RecipeBuilder.new(user_params)
+        if builder.update_call(recipe)
+            render json: { message: 'Recipe updated successfully' }, status: :created
+        else
+            render json: { error: 'Recipe update failed' }, status: :unprocessable_entity
+        end
+    rescue ActiveRecord::RecordInvalid => e
+        render json: { error: e.message }, status: :unprocessable_entity
+    end
```
Comment (jphill19) [inline app/controllers/api/v1/recipe_builder_controller.rb:25]: I like the update route, it looks good!

File: app/controllers/api/v1/recipes_controller.rb [hunk 2]
```diff
@@ -12,7 +12,13 @@ def index
...
+        if params[:by_location].present?
+            update_ingredients = recipe.update_ingredients_details(params[:by_location])
+            recipe_details = RecipeDetails.new(recipe, update_ingredients)
+        else
+            recipe_details = RecipeDetails.new(recipe)
+        end
         render json: RecipeSerializer.format_recipe_details(recipe_details)
     end
+
 end
```
Comment (jphill19) [inline app/controllers/api/v1/recipes_controller.rb:24]: nice update to the show to search for the :by_location

**Review by jphill19 — APPROVED** [review]
This looks really good, I like the updates and incorporates to the testing and routes

**Pull request merged**